js/style/style.js

```diff
@@ -22,8 +22,9 @@ var getWorkerPool = require('../global_worker_pool');
 
 module.exports = Style;
 
-function Style(stylesheet, animationLoop, options) {
-    this.animationLoop = animationLoop || new AnimationLoop();
+function Style(stylesheet, map, options) {
+    this.map = map;
```

Comment:
I've been trying to reduce or eliminate the dependency of other class on `Map`. The dependency here is required by `Source#on{Add,Remove}(map)` -- we should work towards eliminating those two methods.